src/Diagnostics.RuntimeHost/Services/CacheService/DiagEntityTableCacheService.cs

```diff
@@ -16,11 +17,34 @@ public class DiagEntityTableCacheService : IDiagEntityTableCacheService
 
         private IStorageService storageService;
 
+        int cacheExpirationTimeInSecs = 60;
+
         public DiagEntityTableCacheService(IStorageService service)
         {
             cache = new ConcurrentDictionary<string, List<DiagEntity>>();
             storageService = service;
+            StartPolling();
         }
+
+        private async void StartPolling()
+        {
+            do
+            {
+                await Task.Delay(cacheExpirationTimeInSecs * 1000);
+                DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(DiagEntityTableCacheService), "Polling Azure Storage for refreshing cache");
+                var detectorResult = await storageService.GetEntitiesByPartitionkey("Detector");
+                if (detectorResult != null)
+                {
+                    AddOrUpdate("Detector", detectorResult);
+                }
+                var gistResult = await storageService.GetEntitiesByPartitionkey("Gist");
+                if(gistResult != null)
+                {
+                    AddOrUpdate("Gist", gistResult);
+                }
```

Comment:
I would also add a log statement capturing the completion of the call and how much time taken.